isis/tests/FunctionalTestsIsis2pds.cpp

```diff
@@ -0,0 +1,435 @@
+                          "his=yes",
+                          "hrs=yes",
+                          "minimum=-32768",
+                          "maximum=32767"
+                          };
+
+  UserInterface options(APP_XML, args);
+  try {
+    isis2pds(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to open image: " << e.what() << std::endl;
+  }
+
+  Pvl outputLabel(options.GetFileName("TO"));
+  PvlObject imageObject = outputLabel.findObject("IMAGE");
+
+  EXPECT_EQ(int(outputLabel["LABEL_RECORDS"]), 738);
+  EXPECT_EQ(int(outputLabel["^IMAGE"]), 739);
+
+  EXPECT_DOUBLE_EQ(double(imageObject["OFFSET"]), -2.5001526018641);
+  EXPECT_DOUBLE_EQ(double(imageObject["SCALING_FACTOR"]), 1.0000763009309);
+  EXPECT_EQ(int(imageObject["SAMPLE_BITS"]), 16);
+  EXPECT_EQ(QString(imageObject["SAMPLE_BIT_MASK"]), "2#1111111111111111#");
+  EXPECT_EQ(QString(imageObject["SAMPLE_TYPE"]), "LSB_INTEGER");
+  EXPECT_EQ(int(imageObject["CORE_NULL"]), -32768);
+  EXPECT_EQ(int(imageObject["CORE_LOW_REPR_SATURATION"]), -32767);
+  EXPECT_EQ(int(imageObject["CORE_LOW_INSTR_SATURATION"]), -32766);
+  EXPECT_EQ(int(imageObject["CORE_HIGH_REPR_SATURATION"]), -32764);
+  EXPECT_EQ(int(imageObject["CORE_HIGH_INSTR_SATURATION"]), -32765);
+
+}
+
+TEST_F(SmallCube, FunctionalTestIsis2pdsManualu16bit) {
+  QTemporaryDir prefix;
+  QString outputFile = prefix.path() + "/outTemp.img";
+  QVector<QString> args = {"from=" + testCube->fileName(),
+                          "to=" + outputFile,
+                          "stretch=manual",
+                          "bittype=u16bit",
+                          "lrs=yes",
+                          "lis=yes",
+                          "his=yes",
+                          "hrs=yes",
+                          "minimum=0",
+                          "maximum=65535"
+                          };
+
+  UserInterface options(APP_XML, args);
+  try {
+    isis2pds(options);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to open image: " << e.what() << std::endl;
+  }
+
+  Pvl outputLabel(options.GetFileName("TO"));
+  PvlObject imageObject = outputLabel.findObject("IMAGE");
+
+  EXPECT_EQ(int(outputLabel["LABEL_RECORDS"]), 730);
+  EXPECT_EQ(int(outputLabel["^IMAGE"]), 731);
+
+  EXPECT_DOUBLE_EQ(double(imageObject["OFFSET"]), -3.0002289027926);
+  EXPECT_DOUBLE_EQ(double(imageObject["SCALING_FACTOR"]), 1.0000763009309);
+  EXPECT_EQ(QString(imageObject["SAMPLE_TYPE"]), "LSB_UNSIGNED_INTEGER");
+  EXPECT_EQ(int(imageObject["CORE_NULL"]), 0);
+  EXPECT_EQ(int(imageObject["CORE_LOW_REPR_SATURATION"]), 1);
+  EXPECT_EQ(int(imageObject["CORE_LOW_INSTR_SATURATION"]), 2);
+  EXPECT_EQ(int(imageObject["CORE_HIGH_REPR_SATURATION"]), 65535);
+  EXPECT_EQ(int(imageObject["CORE_HIGH_INSTR_SATURATION"]), 65534);
+
+}
+
+TEST(isis2pdsTest, FunctionalTestIsis2pdsOffset) {
+  QTemporaryDir prefix;
+  QString outputFile = prefix.path() + "/outTemp.img";
+  QVector<QString> args = {"from=data/isis2pds/ldem_4_reduced.cub",
```

Comment:
I might've missed a discussion about this at the standups, but do we want to programmatically create the input cubes with a fixture or add these cub files to the test data like you are doing now?

Reply:
My understand from being trained is that if the fixtures do not handle the specific instances that we can alternatively reduce the image that was originally being tested with. For this, it mentioned testing an offset but I was not sure how to test offset with a fixture cube. Therefore I just reduced the original tested cube.

Reply:
I think the offset is referring to the offset keyword in the output image. You could create a new fixture in a similar way to the [Apollo cube](https://github.com/USGS-Astrogeology/ISIS3/blob/936e6fcfa91ace80414c7ac89be1fecf5c52e163/isis/tests/Fixtures.cpp#L706). @Kelvinrr What are your thoughts?

Reply:
I am still having a difficulty finding the correlation between the output offset keywords from the original cub file. It does not seem like the data is being directly pulled from the cube label, therefore I am not sure how to add the correct information to a new fixture.

Reply:
We've pushed small .cubs to the repo for hicubeit, demprep, and ringsautomos.  We've also programmatically created cubes with pvl/isds in footprintinit, apolloNetwork, etc.  It seems like there's no clear policy related to this issue, so I'm of the opinion that this is not a merge-stopping issue.

I think that this is something that we should discuss during standup.

Reply:
I agree. There seems to be multiple ways of testing and we should come up with a standard.